scripts/sanity_chk/harness.py

```diff
@@ -0,0 +1,62 @@
+import re
+from collections import OrderedDict
+
+class Harness:
+    def __init__(self):
+        self.state = None
+        self.type = None
+        self.regex = []
+        self.matches = OrderedDict()
+        self.ordered = True
+        self.repeat = 1
+
+    def configure(self, instance):
+        config = instance.test.harness_config
+        if config:
+            self.type = config.get('type', None)
+            self.regex = config.get('regex', [] )
+            self.repeat = config.get('repeat', 1)
+            self.ordered = config.get('ordered', True)
+
+class Console(Harness):
+
+    def handle(self, line):
+        if self.type == "one_line":
+            pattern = re.compile(self.regex[0])
+            if pattern.match(line):
```

Comment:
As commented in IRC:
This will only match the regular expression if the line does not have any prefix or postfix, which for some targets may be the case (for example a tracing tool may add a time stamp to each line of a trace). It would be nice if there was a `.*` added at the beginning and end of the expression.
So the line above would become ```pattern = re.compile(".*" + self.regex[0] + ".*")``` and the for multi-line ```pattern = re.compile(".*" + r + ".*")```

Otherwise, it would be nice to have at some point a more generic Handler mechanism. So that we could easily define that a given board has a different handler. Then we could ensure that the stream that comes out of the handler into this harness would be already free of extra characters.

Reply:
would re.search() also accomplish this?

Reply:
I think you are right